client/services/SolutionHandlerService.js

```diff
@@ -45,19 +45,28 @@ tie.factory('SolutionHandlerService', [
           // Otherwise, the code doesn't have any obvious syntax errors.
           // Generate a CodeSubmission object that wraps the student's code
           // into a class and appends some test code, then run the whole thing.
+          var correctnessTests = [];
+          var buggyOutputTests = [];
+          var performanceTests = [];
+          for (var i = 0; i < tasks.length; i++) {
+            correctnessTests.push(tasks[i].getCorrectnessTests());
+            buggyOutputTests.push(tasks[i].getBuggyOutputTests());
+            performanceTests.push(tasks[i].getPerformanceTests());
+          }
+          var lastTask = tasks[tasks.length - 1];
           var codeSubmission = CodeSubmissionObjectFactory.create(
             studentCode.trim());
           CodePreprocessorDispatcherService.preprocess(
             language, codeSubmission, auxiliaryCode,
-            task.getInputFunctionName(), task.getMainFunctionName(),
-            task.getOutputFunctionName(), task.getCorrectnessTests(),
-            task.getBuggyOutputTests(), task.getPerformanceTests());
+            lastTask.getInputFunctionName(), lastTask.getMainFunctionName(),
```

Comment:
Oh, I see. But then what is the point of passing in the mainFunctionName at all?

To be clear about what I'm asking, the mainFunctionName for https://github.com/google/tie/blob/master/assets/questions/internationalization.js#L76 starts off as 'abbreviate' but then changes to 'are_all_unique'. In your implementation, I believe there is supposed to be nothing special about the last task. So either the mainFunctionName etc. matters (in which case you'll need the data from all the tasks) or it doesn't matter (in which case you wouldn't need to pass it in at all). Am I missing something here?